### PR TITLE
Microverse Rails Capstone Project: Budget App

### PR DESCRIPTION
**Pull request by peter-akhigbe**
# Changes Made

- Implemented a mobile web application for budget management following the provided design guidelines.
- Created models based on the ERD diagram with necessary validations and tests.
- Incorporated authentication and authorization using the Devise gem.
- Designed and implemented pages for adding categories and transactions.
- Developed the home page displaying categories and their total transaction amounts.
- Ensured the transactions page shows a list of transactions for a selected category, ordered by the most recent.
- Implemented thorough testing, including unit and integration tests for critical components.
- Utilized Postgres as the database and deployed the application online.
- Recorded a 5-minute video presenting the project's features, including a demo and code highlights.

## Additional Notes

- The project adheres to best practices for Ruby and follows correct gitflow.
- Linter errors have been addressed.
- Documentation is provided in a professional manner.
- The README includes appropriate credit to the original design author, Gregoire Vella, as required by the Creative Commons license.

### CODE REVIEW

**Comment by peter-akhigbe** [comment]
# [Presentation Video](https://www.loom.com/share/15eda7c30e5c4f01a0f649eff9dc6b0a?sid=ab751d70-3158-4eae-a23c-163fd3882e97)

File: README.md [hunk 1]
```diff
@@ -1,24 +1,186 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+* I would like to give Microverse my sincerest gratitude for accommodating me in the Full-time Software Development program.
+* [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (DammyShittu) [inline README.md:178]: - [ ] Kindly include the credits to Gregoire Vella the designer.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,186 @@
-# README
+# Microverse Rails Capstone
```
Comment (DammyShittu) [inline README.md:2]: - [ ] You were required to deploy your app and add the live demo link to the readme. Kindly do that.

- [ ] Kindly make a maximum of 5 minutes length presentation of your project talking about the features and the feature you find interesting

File: spec/models/entity_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# /spec/models/entity_spec.rb
+require 'rails_helper'
+
```
Comment (DammyShittu) [inline spec/models/entity_spec.rb:3]: - [ ] Good job writing passing model and controller tests, however, you still need to write integration tests also for integral parts of your app.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @peter-akhigbe,

## Highlights
✔️ Back buttons to correct pages
✔️ Splash screen feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by peter-akhigbe** [comment]
Hello @DammyShittu, thanks for the great review, i have mad the following changes as you suggested;

- Add link to the live website in the README
- Add link to Presentation Video in the README
- Add credit to the designer in the README
- Add integration test for User authentication with RSpec

**Review by binhussen — APPROVED** [review]
Hi @peter-akhigbe , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
IApproveSimonCowellGIF (6)

## Highlights
- Built budget app ✔️
- Well Stractured Code ✔️
- Added Integration Tests ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**